public/index.html

```diff
@@ -13,16 +13,25 @@
         </header>
         <section ng-controller="MainController">
             Remaining tasks: {{ incomplete }}
+           <div>
+                Display
+                <input type="radio" value="all" ng-model="displayTodos" checked="checked">All
+                <input type="radio" value="active" ng-model="displayTodos">Active
+                <input type="radio" value="completed" ng-model="displayTodos">Completed
+                <button ng-hide="complete === 0" ng-click="deleteCompleted()">Delete completed</button>
+            </div>
             <div ng-repeat="todo in todos">
-                <span ng-class="{completeItem: todo.isComplete}">{{ todo.title }}</span>
-                <button ng-click="deleteTodo(todo)">Delete</button>
-                <button ng-click="completeTodo(todo)">Complete</button>
+                <span ng-hide="hideTodo(todo)">
+                    <span ng-class="{completeItem: todo.isComplete}">{{ todo.title }}</span>
+                    <button ng-click="deleteTodo(todo)">Delete</button>
+                    <button ng-click="completeTodo(todo)">Complete</button>
+                </span>
             </div>
             <form id="todo-form">
                 <input id="new-todo" autofocus ng-model="title">
                 <button id="submit-todo" type="submit" ng-click="createTodo()">Create</button>
             </form>
-            <div id="error">{{ errorMessage }}</div>
+            <div id="error" ng-hide="errorMessage === ''">{{ errorMessage }}</div>
```

Comment:
A neat trick you can do here is instead of ng-hide="errorMessage === '', you can do ng-show="errorMessage".